### DISCUSSION TITLE
Getting possible erroneous SAWarning

### DISCUSSION THREAD

**Discussion by radiantone** [sqlalchemy/sqlalchemy#7485] [Usage Questions]
Hi,
  So I have a simple model with relation between two classes PlugModel, ArgumentModel such that I want to associate one ArgumentModel with a PlugModel. I've done this in other models just fine, but running into a posssible bug here.

Given a session with a plug = PlugModel(...) queried from the database and an argument = ArgumentModel() created from scratch I want to:

plug.argument = argument

However, when I commit I get this message
```SAWarning: SELECT statement has a cartesian product between FROM element(s) "plugs_arguments" and FROM element "argument".  Apply join condition(s) between each element to resolve.```

And when I check plug.argument it is always None, even though argument is a valid ArgumentModel

What could be wrong here?
```py
plugs_arguments = Table('plugs_arguments', Base.metadata,
                        Column('plug_id', ForeignKey(
                            'plug.id'), primary_key=True),
                        Column('argument_id', ForeignKey(
                            'argument.id'), primary_key=True)
                        )

class ArgumentModel(BaseModel):
    __tablename__ = 'argument'

    plugs = relationship(
        "PlugModel", secondary=plugs_arguments)
                                
class PlugModel(BaseModel):
    __tablename__ = 'plug'

    argument = relationship(
        "ArgumentModel", secondary=plugs_arguments, back_populates="plugs", uselist=False)
 ```

**Comment by radiantone** [comment]
And there is a relation in the secondary table too, but it is never pulled in when I retrieve the plug.

**Comment by zzzeek** [comment]
seeing that warning during a flush would have to mean some range of uncommon programming patterns are in use, such as mappings against joins/select statements, or relationships with elaborate join conditions that aren't marked viewonly, or event handlers doing things.    i wouldnt be able to provide any insight without details provided  (stack trace, MCVE).

**Comment by radiantone** [comment]
Here is a code fragment that isn't working. I don't get the SAWwarning anymore.

I tried a simpler model design and the problem persists but only for ArgumentModel class. My other classes work fine and load as needed.

Model
```
class PlugModel(BaseModel):
    __tablename__ = 'plug'
    argument_id = Column(String, ForeignKey('argument.id'))
    argument = relationship("ArgumentModel", lazy=True)
```
Code 
```
    _plug = context.obj['database'].session.query(
        PlugModel).filter_by(name=plug).first()

    # First time should be None of course, but subsequent queries should return the related object
    logging.info("Before Plug argument: %s", _plug.argument)
    _argument = ArgumentModel(
        name='test', position=0, kind=1)

    _plug.argument = _argument
    context.obj['database'].session.add(_plug)
    context.obj['database'].session.add(_argument)

    context.obj['database'].session.commit()
    
    # This always prints the related object just fine
    logging.info("After Plug argument: %s", _plug.argument)

```
Here is output
```
2021-12-21 12:09:20,371 : INFO : Before Plug argument: None
2021-12-21 12:09:20,391 : INFO : After Plug argument: <pyfi.db.model.models.ArgumentModel object at 0x7f1826671b80>
```

Even after I run it multiple times the related argument is always None, but argument_id is set.

**Reply by zzzeek** [reply]
hey there -

unfortunately that's not an [MCVE](http://stackoverflow.com/help/mcve) nor is there any stack trace given (SQL log output showing the query that's wrong also would show a lot), and there's additionally nothing unusual about any of that code.

**Reply by radiantone** [reply]
Ok, let me work up something and post it here.
